lib/basedriver/commands/find.js

```diff
@@ -273,29 +272,33 @@ helpers.findByImage = async function findByImage (b64Template, {
     }
   }
 
-  if (!rect) {
+  if (results.length === 0) {
     if (multiple) {
       return [];
     }
     throw new errors.NoSuchElementError();
   }
 
-  log.info(`Image template matched: ${JSON.stringify(rect)}`);
-  if (b64Matched) {
-    log.info(`Matched base64 data: ${b64Matched.substring(0, 200)}...`);
-  }
-  const imgEl = new ImageElement(b64Template, rect, score, b64Matched);
+  const elements = results.map((result) => {
+    log.info(`Image template matched: ${JSON.stringify(result.rect)}`);
+    const b64Matched = result.visualization;
+    if (b64Matched) {
+      log.info(`Matched base64 data: ${b64Matched.substring(0, 200)}...`);
+    }
+    return new ImageElement(b64Template, result.rect, result.score, b64Matched);
+  });
 
   // if we're just checking staleness, return straightaway so we don't add
   // a new element to the cache. shouldCheckStaleness does not support multiple
   // elements, since it is a purely internal mechanism
   if (shouldCheckStaleness) {
-    return imgEl;
+    return elements[0];
   }
 
-  const protocolEl = this.registerImageElement(imgEl);
+  const registeredElements = elements.map((imgEl) => this.registerImageElement(imgEl));
+  log.info(`Registered elements: ${JSON.stringify(registeredElements)}`);
```

Comment:
there is no need to print all these details into the log

Reply:
Sorry, accidentally committed my local debug 😓